zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/RemoteInterpreterEventServer.java

```diff
@@ -557,4 +563,58 @@ public void updateParagraphConfig(String noteId,
       LOGGER.error("Fail to updateParagraphConfig", e);
     }
   }
+
+  @Override
+  public List<LibraryMetadata> getAllLibraryMetadatas(String interpreter) throws TException {
+    if (StringUtils.isBlank(interpreter)) {
+      LOGGER.warn("Interpreter is blank");
+      return Collections.emptyList();
+    }
+    File interpreterLocalRepo = new File(
+        zConf.getAbsoluteDir(ZeppelinConfiguration.ConfVars.ZEPPELIN_DEP_LOCALREPO)
+            + File.separator
+            + interpreter);
+    if (!interpreterLocalRepo.exists()) {
+      LOGGER.warn("Local interpreter repository {} for interpreter {} doesn't exists", interpreterLocalRepo,
+          interpreter);
+      return Collections.emptyList();
+    }
+    if (!interpreterLocalRepo.isDirectory()) {
+      LOGGER.warn("Local interpreter repository {} is no folder", interpreterLocalRepo);
+      return Collections.emptyList();
+    }
+    Collection<File> files = FileUtils.listFiles(interpreterLocalRepo, new String[] { "jar" }, false);
```

Comment:
Only jar is transferred ?

Reply:
Yes, that is correct, is there anything else to be transferred?
Currently, I only have Jar artefacts in the packages section of the interpreters, which are downloaded with Maven.

Reply:
In most of cases it should be OK, Just concern some corner cases that some library may depend on non-jar files.

Reply:
I can add more extensions, but at the moment I don't know which ones.

Reply:
Yeah, I think it should be OK for now.